File: exampleProjects/PythonExample/src/de/sots/cellarsandcentaurs/domain/models/armor_class.py
from de.sots.cellarsandcentaurs.application.creature_util import CreatureUtil

class ArmorClass:
    def __init__(self, base, bonus, description=CreatureUtil.STANDARD_ARMOR_CLASS_DESCRIPTION):
        self.description = description
        self.base = base
        self.bonus = bonus

    @property
    def total(self):
        return self.base + self.bonus

    @property
    def base(self):
        return self._base

    @base.setter
    def base(self, value):
        self._base = value

    @property
    def bonus(self):
        return self._bonus

    @bonus.setter
    def bonus(self, value):
        self._bonus = value

    @property
    def description(self):
        return self._description

    @description.setter
    def description(self, value):
        self._description = value

File: exampleProjects/PhpExample/src/de/sots/cellarsandcentaurs/domain/model/ArmorClass.php
<?php

namespace de\sots\cellarsandcentaurs\domain\model;

use de\sots\cellarsandcentaurs\application\CreatureUtil;

class ArmorClass
{
    private string $description;
    private int $base;
    private int $bonus;
    private int $total;

    public function __construct(int $base, int $bonus, ?string $description = null)
    {
        $this->description = $description ?? CreatureUtil::STANDARD_ARMOR_CLASS_DESCRIPTION;
        $this->base = $base;
        $this->bonus = $bonus;
        $this->total = $base + $bonus;
    }

    public function getBase(): int
    {
        return $this->base;
    }

    public function setBase(int $base): void
    {
        $this->base = $base;
        $this->updateTotal();
    }

    public function getBonus(): int
    {
        return $this->bonus;
    }

    public function setBonus(int $bonus): void
    {
        $this->bonus = $bonus;
        $this->updateTotal();
    }

    public function getTotal(): int
    {
        return $this->total;
    }

    public function getDescription(): string
    {
        return $this->description;
    }

    public function setDescription(string $description): void
    {
        $this->description = $description;
    }

    private function updateTotal(): void
    {
        $this->total = $this->base + $this->bonus;
    }
}
